### ISSUE TITLE
doom/sudo-this-file does not work, neither does /sudo: (tramp)

### ISSUE THREAD

**Issue by t-e-r-m** [hlissner/doom-emacs#5363]
### What did you expect to happen?

Hi, I'm using Emacs HEAD (native-comp + pgtk) and the latest commit of Doom Emacs.

I tried to edit a file as root and got an error message.

It has something to do with vertico and tramp, because manually using `tramp` does not resolve this issue.

### What actually happened?

Backtrace after `SPC f U`

```
Debugger entered--Lisp error: (void-variable tramp-security-key-confirm-regexp)
  signal(void-variable (tramp-security-key-confirm-regexp))
  tramp-maybe-open-connection((tramp-file-name "sudo" "root" nil "localhost" nil "/home/myuser/.emacs.d/.local/straight/repos" nil))
  tramp-send-command((tramp-file-name "sudo" "root" nil "localhost" nil "/home/myuser/.emacs.d/.local/straight/repos" nil) "test 0 2>/dev/null; echo tramp_exit_status $?")
  tramp-send-command-and-check((tramp-file-name "sudo" "root" nil "localhost" nil "/home/myuser/.emacs.d/.local/straight/repos" nil) "test 0")
  tramp-get-test-command((tramp-file-name "sudo" "root" nil "localhost" nil "/home/myuser/.emacs.d/.local/straight/repos" nil))
  tramp-run-test("-d" "/sudo:root@localhost:/home/myuser/.emacs.d/.local/st...")
  tramp-sh-handle-file-directory-p("/sudo:root@localhost:/home/myuser/.emacs.d/.local/st...")
  tramp-sh-file-name-handler(file-directory-p "/sudo:root@localhost:/home/myuser/.emacs.d/.local/st...")
  apply(tramp-sh-file-name-handler file-directory-p "/sudo:root@localhost:/home/myuser/.emacs.d/.local/st...")
  tramp-file-name-handler(file-directory-p "/sudo:root@localhost:/home/myuser/.emacs.d/.local/st...")
  file-directory-p("/sudo:root@localhost:/home/myuser/.emacs.d/.local/st...")
  find-file-noselect("/sudo:root@localhost:/home/myuser/.emacs.d/.local/st..." nil nil nil)
  find-file("/sudo:root@localhost:/home/myuser/.emacs.d/.local/st...")
  doom/sudo-this-file()
  funcall-interactively(doom/sudo-this-file)
  command-execute(doom/sudo-this-file)
```

### Steps to reproduce

1. Open a file/dir
2. `SPC f U`
3. (you now get an error)

Alternatively, try Tramp in `SPC f f` for the same result.

### System Information

https://0x0.st/-JMF.txt

**Comment by hlissner** [comment]
> packages {omitted}

Please don't omit things in the your `M-x doom/info`. Your installed packages is _exactly_ what I needed to know in this case; in particular, if you're installing a newer version of tramp yourself, as some folks commonly do (and shouldn't).

So, are you installing tramp?

-----

In any case, I'll mark this as unknown since I can't reproduce this native-comp + pgtk + vertico on vanilla Doom.

**Comment by carlosperezm** [comment]
Hi

I am having exactly the same issue you are describing here.
It seems to be independent of Vertico, because I am using Ivy

### Backtrace:

```
Debugger entered--Lisp error: (void-variable tramp-security-key-confirm-regexp)
  signal(void-variable (tramp-security-key-confirm-regexp))
  tramp-maybe-open-connection((tramp-file-name "sudo" "root" nil "localhost" nil "/etc/pacman.conf" nil))
  tramp-send-command((tramp-file-name "sudo" "root" nil "localhost" nil "/etc/pacman.conf" nil) "test 0 2>/dev/null; echo tramp_exit_status $?")
  tramp-send-command-and-check((tramp-file-name "sudo" "root" nil "localhost" nil "/etc/pacman.conf" nil) "test 0")
  tramp-get-test-command((tramp-file-name "sudo" "root" nil "localhost" nil "/etc/pacman.conf" nil))
  tramp-run-test("-d" "/sudo:root@localhost:/etc/pacman.conf")
  tramp-sh-handle-file-directory-p("/sudo:root@localhost:/etc/pacman.conf")
  tramp-sh-file-name-handler(file-directory-p "/sudo:root@localhost:/etc/pacman.conf")
  apply(tramp-sh-file-name-handler file-directory-p "/sudo:root@localhost:/etc/pacman.conf")
  tramp-file-name-handler(file-directory-p "/sudo:root@localhost:/etc/pacman.conf")
  file-directory-p("/sudo:root@localhost:/etc/pacman.conf")
  find-file-noselect("/sudo:root@localhost:/etc/pacman.conf" nil nil nil)
  find-file("/sudo:root@localhost:/etc/pacman.conf")
  doom/sudo-this-file()
  #<subr funcall-interactively>(doom/sudo-this-file)
  apply(#<subr funcall-interactively> doom/sudo-this-file)
  funcall-interactively(doom/sudo-this-file)
  #<subr call-interactively>(doom/sudo-this-file record nil)
  apply(#<subr call-interactively> (doom/sudo-this-file record nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> doom/sudo-this-file record nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (doom/sudo-this-file record nil))
  call-interactively(doom/sudo-this-file record nil)
  command-execute(doom/sudo-this-file record)
  counsel-M-x-action("doom/sudo-this-file")
  ivy-call()
  ivy-read("M-x " ("doom/info" "doom-debug-mode" "tramp-recompile-elpa" "tramp-cleanup-all-connections" "tramp-version" "+vterm/here" "pdf-tools-install" "check-parens" "langtool-correct-buffer" "projectile-invalidate-cache" "langtool-check" "lsp-install-server" "doom/sudo-find-file" "langtool-check-done" "eshell" "org-mode" "load-theme" "emacs-version" "spell-fu-mode" "doom/sudo-this-file" "langtool-switch-default-language" "cd" "5x5" "amx" "arp" "dbx" "dig" "erc" "ert" "eww" "ftp" "gdb" "irc" "jdb" "lsp" "man" "mpc" "pdb" "pwd" "rsh" "sdb" "xdb" "calc" "deft" "diff" "dirs" "ffap" "gnus" "grep" "help" ...) :predicate counsel--M-x-externs-predicate :require-match t :history counsel-M-x-history :action counsel-M-x-action :keymap (keymap (67108908 . counsel--info-lookup-symbol) (67108910 . counsel-find-symbol)) :initial-input nil :caller counsel-M-x)
  counsel-M-x()
  #<subr funcall-interactively>(counsel-M-x)
  apply(#<subr funcall-interactively> counsel-M-x)
  funcall-interactively(counsel-M-x)
  #<subr call-interactively>(counsel-M-x nil nil)
  apply(#<subr call-interactively> (counsel-M-x nil nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> counsel-M-x nil nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (counsel-M-x nil nil))
  call-interactively(counsel-M-x nil nil)
  command-execute(counsel-M-x)
```

### System information

https://pastebin.com/iq2qGNDy

**Comment by t-e-r-m** [comment]
> > packages {omitted}
>
> Please don't omit things in your `M-x doom/info`. Your installed packages is _exactly_ what I needed to know in this case; in particular, if you're installing a newer version of tramp yourself, as some folks commonly do (and shouldn't).
> So, are you installing tramp?

Sorry, I'll include the installed packages next time.  I had not separately installed Tramp.

Like carlosperezm, I could also reproduce it in Vanilla Emacs.

I upgraded Emacs and the issue has been fixed for me, I can now use Tramp without any issues on both Vanilla Emacs and Doom Emacs.

Related Commits:

https://github.com/emacs-mirror/emacs/commit/fd2d56d63c8a9dbc5a96d2eb746761b2ecc2c573

https://github.com/emacs-mirror/emacs/commit/e5b2e5356f78e70fbb3dbfdb2abff17bb84efc2d

I'll close this issue now, this was an Emacs bug that seems to be fixed now.
